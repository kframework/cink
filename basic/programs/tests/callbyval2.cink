int f(int x, bool b) {
  if (b) return x * 2;
  return x * 3;
}

int main() {
  cout << f(5, false);
  return 0;
}

