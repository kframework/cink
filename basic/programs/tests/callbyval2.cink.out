15<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            f |-> #symLoc(1)
            main |-> #symLoc(2)
        </env>
    </control>
    <genv>
        f |-> #symLoc(1)
        main |-> #symLoc(2)
    </genv>
    <store>
        #symLoc(1) |-> lambda ( ((int x), (bool b)) @ ((if ( b ) (return (
          x * 2) ;) else { }) (return (x * 3) ;)) )
        #symLoc(2) |-> lambda ( .Decls @ (((cout << (f ( (5 , false) ))) ;) (
          return 0 ;)) )
        #symLoc(3) |-> 5
        #symLoc(4) |-> false
    </store>
</T>
